holidays/countries/cayman_islands.py

```diff
@@ -0,0 +1,166 @@
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays import APR, MAY, JUN, SEP, NOV
+from holidays.groups import ChristianHolidays, InternationalHolidays, StaticHolidays
+from holidays.observed_holiday_base import (
+    ObservedHolidayBase,
+    SAT_SUN_TO_NEXT_MON,
+    SAT_SUN_TO_NEXT_MON_TUE,
+)
+
+
+class CaymanIslands(ObservedHolidayBase, ChristianHolidays, InternationalHolidays, StaticHolidays):
+    """Cayman Islands holidays.
+
+    References:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_the_Cayman_Islands>
+        * <https://web.archive.org/web/20250212214050/https://www.timeanddate.com/holidays/cayman-islands/2025>
+        * [Public Holidays Law (2007 Revision)](https://web.archive.org/web/20250227060525/https://legislation.gov.ky/cms/images/LEGISLATION/PRINCIPAL/1964/1964-0140/PublicHolidaysAct_2007%20Revision_g.pdf)
+        * [Public Holidays Order, 2024](https://web.archive.org/web/20240518181823/https://legislation.gov.ky/cms/images/LEGISLATION/AMENDING/2024/2024-O004/PublicHolidaysOrder2024SL4of2024.pdf)
+        * [Public Holidays Order, 2025](https://legislation.gov.ky/cms/images/LEGISLATION/SUBORDINATE/2025/2025-0015/PublicHolidaysOrder2025_SL%2015%20of%202025.pdf)
+        * [2021](https://www.gov.ky/news/press-release-details/public-holidays-for-2021)
+        * [2022](https://www.gov.ky/news/press-release-details/public-holidays-2022)
+        * [2025](https://www.gov.ky/calendar/public-holidays)
+    """
+
+    country = "KY"
+    default_language = "en_GB"
+    # %s observed.
+    observed_label = tr("%s (observed)")
+    # Public Holidays Law (2007 Revision).
+    start_year = 2008
+    supported_languages = ("en_GB", "en_US")
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        StaticHolidays.__init__(self, cls=CaymanIslandsStaticHolidays)
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_observed(self._add_new_years_day(tr("New Year's Day")))
+
+        # National Heroes Day.
+        self._add_holiday_4th_mon_of_jan(tr("National Heroes Day"))
+
+        # Ash Wednesday.
+        self._add_ash_wednesday(tr("Ash Wednesday"))
+
+        # Good Friday.
+        self._add_good_friday(tr("Good Friday"))
+
+        # Easter Monday.
+        self._add_easter_monday(tr("Easter Monday"))
+
+        general_election_dates = {
+            2009: (MAY, 20),
+            2017: (MAY, 24),
+            2021: (APR, 14),
+            2025: (APR, 30),
+        }
+        if self._year in general_election_dates:
+            # General Election Day.
+            self._add_holiday(tr("General Election Day"), general_election_dates[self._year])
```

Comment:
Considering the fact that there's no fixed pattern (i.e. Nth Day of Xth Month every Y year) for this, IMO it might be better to move these to the `CaymanIslandsStaticHolidays` section